api/src/infrastructure/kubernetes/infrastructure.rs

```diff
@@ -478,6 +495,136 @@ impl KubernetesInfrastructure {
 
         Ok(service)
     }
+
+    async fn create_persistent_volume_claim<'a>(
+        &self,
+        app_name: &str,
+        service: &'a DeployableService,
+    ) -> Result<HashMap<&'a String, PersistentVolumeClaim>, KubernetesInfrastructureError> {
```

Comment:
I'm thinking that expressing that there is sometime not a value is better.

```suggestion
    ) -> Result<Option<HashMap<&'a String, PersistentVolumeClaim>>, KubernetesInfrastructureError> {
```

Reply:
Directly implemented Option here instead of Result